balancer-js/src/modules/joins/joins.module.ts

```diff
@@ -837,7 +924,21 @@ export class Join {
       fromInternalBalance: sender === this.relayer,
     });
 
-    return call;
+    const amountsTokenIn = sortedAmounts.map((a) =>
+      Relayer.isChainedReference(a) ? '0' : a
+    );
+    const amountOut = Relayer.isChainedReference(minAmountOut)
+      ? '0'
+      : minAmountOut;
+
+    return [
+      call,
+      sender === this.relayer ? [] : sortedTokens,
```

Comment:
Same here about comments giving context to why values are empty if sender is the relayer.